### ISSUE TITLE
uri module incompatible with debian httplib2 (fixed in wheezy)

### ISSUE THREAD

**Issue by filipncs** [ansible/ansible#2291]
The uri module uses the _disable_ssl_certificate_validation_ parameter of the httplib2.Http constructor. This is not supported in the httplib2 available in the stable version of debian (squeeze). It is however supported in the version currently in testing (wheezy).

Could we bundle our own httplib2 or not use the _disable_ssl_certificate_validation_ parameter when the current machine has an older version of the library? Simple testing indicates that the module is still quite useful even when _disable_ssl_certificate_validation_ isn't activated.

**Comment by rrtj3** [comment]
On Sun, Mar 3, 2013 at 11:52 PM, Filip Svendsen notifications@github.comwrote:

> The uri module uses the _disable_ssl_certificate_validation_ parameter of
> the httplib2.Http constructor. This is not supported in the httplib2
> available in the stable version of debian (squeeze). It is however
> supported in the version currently in testing (wheezy).

I'll see if I can determine when this feature was added, but what version
of httplib2 is squeeze running?

> Could we bundle our own httplib2 or not use the *
> disable_ssl_certificate_validation\* parameter when the current machine
> has an older version of the library?
> 
> I don't think bundling our own httplib2 is a good idea. If we can cleanly
> determine if the 'disable_ssl_certificate_validation' feature is available
> or not I don't see a problem with doing that and noting it in the
> documentation. We'll probably want to enable the 'ca_certs' parameter at
> the same time though (assuming that is supported in the version on squeeze)
> so people can pass in a custom CA certificates bundle to use, so they can
> get past untrusted certs if needed. I _think_ that's what the 'ca_certs'
> parameter is for anyway. Need to determine for sure.
> 
> Simple testing indicates that the module is still quite useful even when *
> disable_ssl_certificate_validation\* isn't activated.

Just out of interest, what are you using the URI module for? (I just like
to know how folks are using it.)

Thanks

**Comment by mpdehaan** [comment]
Bundling our own http lib is unacceptable, just add some try/except code.   Patches would be accepted.

Debian Stable!  Wow!   It's like a piece of history :)

**Comment by gplasky** [comment]
This unfortuantely is still active for those of us stuck on Debian Squeeze. I perused the changelogs and it seems that the SSL verification was added in httplib2 0.7.0, which leaves those of us on Squeeze stuck on 0.6.0. We'll probably roll our own upgraded package however I'd like to see the docs updated to reflect the requirement of 0.7.0 to spare future pain.

We _could_ get fancy and have the module gracefully degrade based on an additional config parameter (e.g. `ssl_validation: (true|false)` where true would invoke the backwards-compatible method signature), but I think that is a less ideal option.
